### ISSUE TITLE
Osu!Lazer crashed after a short blackscreen

### ISSUE THREAD

**Issue by Kiddo-Kris** [ppy/osu#6810]
**Describe the crash:** 
Osu!Lazer get crashed after a short blackscreen
**Screenshots or videos showing encountered issue:** 
sorry,i failed to capture any screenshots and video
**osu!lazer version:** 
app-2019.1106.0
**Logs:** 
[database.log](https://github.com/ppy/osu/files/3831903/database.log)
[network.log](https://github.com/ppy/osu/files/3831904/network.log)
[runtime.log](https://github.com/ppy/osu/files/3831905/runtime.log)
[client.zip](https://github.com/ppy/osu/files/3831946/client.zip)

**Computer Specifications:** 
![TIM截图20191111231203](https://user-images.githubusercontent.com/56265123/68597882-b0919780-04d8-11ea-944d-ec9fc2429c3c.png)

Other Informations
No database imports(Idk what is that)
No beatmaps were deleted recently(since1-2month ago)
Client was reinstalled(Delete [app-xxxx.xxxx.x] and run latest osu installer )
There was a sound driver error that stop osu!lazer loading. solved

**Comment by bdach** [comment]
Please attach your database file (`%AppData%/osu/client.db`).

Have you ran any database imports recently? Deleted any beatmaps? Any extra info if you've done something regarding the DB would be appreciated.

**Comment by Kiddo-Kris** [comment]
> Please attach your database file (%AppData%/osu/client.db).
> Have you ran any database imports recently? Deleted any beatmaps? Upgraded the client from an older version? Any extra info if you've done something regarding the DB would be appreciated.

database file and extra infomations added to the post,ty

**Comment by bdach** [comment]
I confirmed that this occurs after swapping in the attached database file. Looking at the actual DB contents this seems to be similar in cause #6109 - there are scores for a beatmap set (ID 221) that's pending deletion which leads the game to take a foreign key violation when trying to purge beatmaps marked for deletion on startup.

Unsure why this surfaced now, since it should have been fixed already. The last problematic score was set on September 30, which should have been after the fix has deployed - unless the client hasn't been updated regularly...
